Tasks/UseDotNetV2/models.ts

```diff
@@ -121,13 +121,13 @@ export class VersionParts {
         try {
             let parts = version.split('.');
             // validate version
-            if ((parts.length != 3) || // check if the version has 3 parts
+            if ((parts.length < 3) || // check if the version has at least 3 parts
                 !parts[0] || // The major version must always be set
                 !parts[1] || // The minor version must always be set
                 !parts[2] || // The patch version must always be set
                 Number.isNaN(Number.parseInt(parts[0])) || // the major version number must be a number
                 Number.isNaN(Number.parseInt(parts[1])) || // the minor version number must be a number
-                Number.isNaN(Number.parseInt(parts[2].split('-')[0])) // the patch version number must be a number. (the patch version can have a '-' because of version numbers like: 1.0.0-beta-50)
+                Number.isNaN(Number.parseInt(parts[2].split(/\-|\+/)[0])) // the patch version number must be a number. (the patch version can have a '-', or a '+' because of version numbers like: 1.0.0-beta-50)
```

Comment:
Can you point me to such an example where + is used

Reply:
This is to make it semver compliant. https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions
In Dotnetcore releases I think so far they have not used it, but if they do in the future, then this change will be required.